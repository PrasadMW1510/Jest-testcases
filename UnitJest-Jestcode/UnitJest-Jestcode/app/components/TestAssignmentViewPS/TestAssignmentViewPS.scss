.test-assignment-ps__heading {
  font-size: 9pt;
  margin-left: 1%;
  margin-bottom: 1.5%;
}

.test-assignment-ps__content {
  background-color: #fff;
  display: inline-block;
  height: 320px;
  width: 97%;
  margin-left: 1%;
  margin-right: 1%;
  font-size: 8pt;
  margin-top: 3%;
}

.test-assignment-ps__quadrants {
  height: 90%;
  border: 1px solid #000;
}

.test-assignment-ps__options {
  padding-left: 5px;
  font-size: 8pt;
}

.test-assignment-ps__block-spacing {
  padding-top: 40px;
  padding-left: 30px;
}

.test-assignment-ps-text {
  font-weight: bold;
  font-size: 8pt;
  padding-right: 60px;
}

.test-assignment-ps-program-text {
  font-weight: bold;
  font-size: 8pt;
  padding-right: 86px;
}

.test-assignment-ps-module-text {
  font-weight: bold;
  font-size: 8pt;
  padding-right: 55px;
}

.test-assignment-ps-test-text {
  font-weight: bold;
  font-size: 8pt;
  padding-right: 120px;
}

.test-assignment-ps__select-field {
  width: 180pt;
  border-style: outset ;
}

.test-assignment-ps__select-label {
  font-weight: bold;
  font-size: 8pt;
  margin-right: 60px;
}

.test-assignment-ps__select-field-disabled {
  width: 180pt;
  border-style: outset ;
  pointer-events: none;
  cursor: not-allowed;
  border-color: #7d8084;
  border-style: inset;
}

.test-assignment-ps__radio-none {
  display: none;
}

.test-assignment-ps__select-label-test {
  font-weight: bold;
  font-size: 8pt;
  margin-right:110px;
}

.test-assignment-ps__select-test {
  padding-top: 60px;
  padding-left: 30px;
  font-weight: bold;
  font-size: 8pt;
}

.test-assignment-ps__quadrants-heading {
  font-weight: bold;
  vertical-align: middle;
  padding-left: 10px;
  background-color: #ffd400;
  width: 99.9%;
  display: inline-block;
  margin-top: 0%;
  padding-top: 0.5%;
  z-index: +1;
}

.test-assignment-ps__checkbox {
  padding-left: 10px;
  padding-top: 0.5%;
}

.test-assignment-ps__radio {
  padding-left: 10px;
  padding-top: 0.5%;
  padding-right: 25px;
}

.test-assignment-ps__footer {
  width: 99.9%;
  display: inline-block;
  margin-top: 29px;
}
