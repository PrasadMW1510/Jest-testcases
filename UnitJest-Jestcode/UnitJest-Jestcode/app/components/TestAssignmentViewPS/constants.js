export const PROGRAMS = {
  DTM_NOW: 'DTM_NOW',
  DTM_MODULE: 'DTM_MODULE',
};

export const SUBPRODUCT = {
  PRODUCT_NAME: 'Do The Math Now!',
};
