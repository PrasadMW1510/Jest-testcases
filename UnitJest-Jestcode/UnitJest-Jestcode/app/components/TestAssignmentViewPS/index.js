/**
 *
 * TestAssignmentViewPS
 *
 */

export { default } from './TestAssignmentViewPS';
