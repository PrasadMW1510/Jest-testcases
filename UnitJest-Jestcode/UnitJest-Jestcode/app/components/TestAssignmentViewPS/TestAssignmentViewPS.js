/**
 *
 * TestAssignmentViewPS
 *
 */

import PropTypes from 'prop-types';
import React from 'react';
import ProgramSettingsButtons from 'components/ProgramSettingsButtons';
import SettingsSelectBox from 'components/SettingsSelectBox';
import './TestAssignmentViewPS.scss';
import * as Constants from './constants';

class TestAssignmentViewPS extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dtmModulesObj:
        this.props.testAssignmentData.get('dtmModules') &&
        this.props.testAssignmentData.get('dtmModules').toJS(),
      subProductObj:
        this.props.testAssignmentData.get('subproduct') &&
        this.props.testAssignmentData.get('subproduct').toJS(),
      dtmTestsObj:
        this.props.testAssignmentData.get('dtmTests') &&
        this.props.testAssignmentData.get('dtmTests').toJS(),
      dtmModules: [],
      toggleModule: true,
      toggleTest: true,
      selectedTestId: '',
      module_id: '',
      toggleDtmModules: false,
      toggleDtmTests: false,
    };
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.testAssignmentData !== this.props.testAssignmentData) {
      this.setState({
        dtmModulesObj:
          nextProps.testAssignmentData.get('dtmModules') &&
          nextProps.testAssignmentData.get('dtmModules').toJS(),
        subProductObj:
          nextProps.testAssignmentData.get('subproduct') &&
          nextProps.testAssignmentData.get('subproduct').toJS(),
        dtmTestsObj:
          nextProps.testAssignmentData.get('dtmTests') &&
          nextProps.testAssignmentData.get('dtmTests').toJS(),
      });
    }
  }

  getModules() {
    return (
      this.state.dtmModules &&
      this.state.dtmModules.map(option1 => (
        <option key={option1.module_id} value={option1.module_id}>
          {`${option1.topic_name}: ${option1.short_name}`}
        </option>
      ))
    );
  }

  getTests() {
    const selectedTests =
      this.state.dtmTests &&
      this.state.dtmTests.filter(tests => tests.module_id[0] === this.state.module_id);
    return (
      selectedTests &&
      selectedTests.map(tests => (
        <option key={tests.test_id} value={tests.test_id}>
          {`${tests.test_name}`}
        </option>
      ))
    );
  }

  isStudentEnrolled = dtmObj => {
    const totalCount =
      this.state.subProductObj &&
      this.state.subProductObj.sub_product.find(
        subprod => subprod.application_id[0] === dtmObj && Number(subprod.total_students[0]) > 0
      );
    if (totalCount) return false;
    return true;
  };

  handleSave() {}

  handleSetInitialValues = () => {
    this.setState({
      dtmModulesObj:
        this.props.testAssignmentData.get('dtmModules') &&
        this.props.testAssignmentData.get('dtmModules').toJS(),
      subProductObj:
        this.props.testAssignmentData.get('subproduct') &&
        this.props.testAssignmentData.get('subproduct').toJS(),
      dtmTestsObj:
        this.props.testAssignmentData.get('dtmTests') &&
        this.props.testAssignmentData.get('dtmTests').toJS(),
      dtmModules: [],
      toggleModule: true,
      toggleTest: true,
      selectedTestId: '',
      module_id: '',
      application_id: '',
      toggleDtmModules: false,
      toggleDtmTests: false,
    });
    this.props.setIsolateTab(false);
  };

  handleSelectProgramOption = e => {
    this.setState({
      dtmModules:
        e.target.id === Constants.PROGRAMS.DTM_MODULE
          ? this.state.dtmModulesObj && this.state.dtmModulesObj[0].module
          : this.state.dtmModulesObj && this.state.dtmModulesObj[1].module,
      dtmTests:
        e.target.id === Constants.PROGRAMS.DTM_MODULE
          ? this.state.dtmTestsObj && this.state.dtmTestsObj.tests[0].test
          : this.state.dtmTestsObj && this.state.dtmTestsObj.tests[1].test,
      toggleModule: false,
      toggleDtmModules: true,
      module_id: '',
      application_id: e.target.id,
    });
  };
  handleSelectedModule = e => {
    this.setState({
      module_id: e.target.value,
      toggleTest: false,
      toggleDtmModules: true,
      toggleDtmTests: true,
    });
  };

  handleSelectTest = e => {
    this.setState({ selectedTestId: e.target.value, toggleDtmTests: true });
    this.props.setIsolateTab(true);
  };

  checkDtmNowEnrollment = programName => {
    const enrollDtm =
      this.state.subProductObj &&
      this.state.subProductObj.sub_product.find(
        subprod => subprod.application_id[0] === programName && subprod.total_students[0] !== '0'
      );
    if (enrollDtm) {
      return true;
    }
    return false;
  };

  renderTestAssignment() {
    return (
      <div className="test-assignment-ps__quadrants">
        <span className="test-assignment-ps__quadrants-heading">Test Assignment</span>
        <div className="test-assignment-ps__block-spacing">
          <span className="test-assignment-ps-program-text"> 1. Select a Program </span>
          <span
            className={
              this.checkDtmNowEnrollment(Constants.PROGRAMS.DTM_MODULE)
                ? 'test-assignment-ps__radio'
                : 'test-assignment-ps__radio-none'
            }
            key="DTM_MODULE"
          >
            <input
              type="radio"
              name="application_id"
              onChange={this.handleSelectProgramOption}
              id="DTM_MODULE"
              key="DTM_MODULE"
              // value="DTM_MODULE"
              disabled={this.isStudentEnrolled(Constants.PROGRAMS.DTM_MODULE)}
              readOnly
              checked={
                this.state.application_id &&
                this.state.application_id === Constants.PROGRAMS.DTM_MODULE
              }
            />
            <span className="test-assignment-ps__options">Do the Math Modules</span>
          </span>
          <span
            className={
              this.checkDtmNowEnrollment(Constants.PROGRAMS.DTM_NOW)
                ? 'test-assignment-ps__radio'
                : 'test-assignment-ps__radio-none'
            }
            key="DTM_NOW"
          >
            <input
              type="radio"
              name="application_id"
              onChange={this.handleSelectProgramOption}
              id="DTM_NOW"
              key="DTM_NOW"
              value="DTM_NOW"
              disabled={this.isStudentEnrolled(Constants.PROGRAMS.DTM_NOW)}
              readOnly
              checked={
                this.state.application_id &&
                this.state.application_id === Constants.PROGRAMS.DTM_NOW
              }
            />
            <span className="test-assignment-ps__options">Do the Math Now</span>
          </span>
        </div>
        <div className="test-assignment-ps__block-spacing">
          <span>
            {' '}
            <SettingsSelectBox
              labelClass="test-assignment-ps__select-label"
              fieldClass={
                this.state.toggleModule
                  ? 'test-assignment-ps__select-field-disabled'
                  : 'test-assignment-ps__select-field'
              }
              fieldValue={this.state.module_id}
              onChange={this.handleSelectedModule}
              label="2. Select a Module or Unit"
            >
              <option value="" disabled hidden={this.state.toggleDtmModules}>
                {' '}
              </option>
              {this.getModules()}
            </SettingsSelectBox>
          </span>
        </div>
        <div className="test-assignment-ps__select-test">
          <span>
            <SettingsSelectBox
              labelClass="test-assignment-ps__select-label-test"
              fieldClass={
                this.state.toggleTest
                  ? 'test-assignment-ps__select-field-disabled'
                  : 'test-assignment-ps__select-field'
              }
              label="3. Select a Test:"
              fieldValue={this.state.selectedTestId}
              onChange={this.handleSelectTest}
            >
              <option value="" disabled hidden={this.state.toggleDtmTests}>
                {''}
              </option>
              {this.getTests()}
            </SettingsSelectBox>
          </span>
        </div>
      </div>
    );
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <span className="test-assignment-ps__heading" />
          <div className="test-assignment-ps__content">{this.renderTestAssignment()}</div>
          <div className="test-assignment-ps__footer">
            <ProgramSettingsButtons
              showRestoreDefaults={false}
              saveAndReturnHandler={this.handleSave}
              setInitialValuesHandler={this.handleSetInitialValues}
              stateResult={!this.props.isTabIsolated}
            />
          </div>
        </div>
      </form>
    );
  }
}

TestAssignmentViewPS.propTypes = {
  // TODO ADD THE FUNCTIONALITY FOR SAVE AND ISOLATE TAB
  // handleSave: PropTypes.func.isRequired,
  setIsolateTab: PropTypes.func.isRequired,
  isTabIsolated: PropTypes.bool.isRequired,
  testAssignmentData: PropTypes.any.isRequired,
};

export default TestAssignmentViewPS;
