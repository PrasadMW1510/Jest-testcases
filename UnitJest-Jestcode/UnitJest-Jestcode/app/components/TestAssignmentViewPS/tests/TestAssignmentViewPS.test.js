import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { fromJS } from 'immutable';
import TestAssignmentViewPS from '../index';

describe('<TestAssignmentViewPS />', () => {
  let wrapper = null;
  let mockHandleSave = null;
  let mockSetIsolateTab = null;
  let mockIsTabIsolated = null;
  let mockTestAssignmentData = null;

  describe('expect to render ProgramSettingsViewPS', () => {
    beforeEach(() => {
      mockHandleSave = jest.fn();
      mockSetIsolateTab = jest.fn();
      mockIsTabIsolated = false;
      mockTestAssignmentData = fromJS({
        dtmModules: {
          module: {
            topic_name: ['Addition &amp; Subtraction'],
            module_id: ['module_3_4'],
            short_name: ['Module C'],
            long_name: ['Numbers greater than 100'],
          },
        },
        dtmTests: {
          tests: {
            test_id: ['dtm_topic1_module1_2_end'],
            module_id: ['module_0_1'],
            test_name: ['End-of-Module Test'],
            test_topic: ['Addition &amp; Subtraction'],
          },
        },
        subproduct: {
          sub_product: [
            {
              name: ['Do The Math Now!'],
              application_id: ['DTM_NOW'],
              total_students: ['0'],
            },
          ],
        },
      });

      wrapper = shallow(
        <TestAssignmentViewPS
          handleSave={mockHandleSave}
          setIsolateTab={mockSetIsolateTab}
          testAssignmentData={mockTestAssignmentData}
          isTabIsolated={mockIsTabIsolated}
        />
      );
    });
    it('Expect to render correctly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
