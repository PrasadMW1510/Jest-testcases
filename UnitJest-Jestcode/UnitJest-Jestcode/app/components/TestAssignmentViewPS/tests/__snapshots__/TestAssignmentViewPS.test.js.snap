// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestAssignmentViewPS /> expect to render ProgramSettingsViewPS Expect to render correctly 1`] = `
<form>
  <div>
    <span
      className="test-assignment-ps__heading"
    />
    <div
      className="test-assignment-ps__content"
    >
      <div
        className="test-assignment-ps__quadrants"
      >
        <span
          className="test-assignment-ps__quadrants-heading"
        >
          Test Assignment
        </span>
        <div
          className="test-assignment-ps__block-spacing"
        >
          <span
            className="test-assignment-ps-program-text"
          >
             1. Select a Program 
          </span>
          <span
            className="test-assignment-ps__radio-none"
            key="DTM_MODULE"
          >
            <input
              disabled={true}
              id="DTM_MODULE"
              key="DTM_MODULE"
              name="application_id"
              onChange={[Function]}
              readOnly={true}
              type="radio"
            />
            <span
              className="test-assignment-ps__options"
            >
              Do the Math Modules
            </span>
          </span>
          <span
            className="test-assignment-ps__radio-none"
            key="DTM_NOW"
          >
            <input
              disabled={true}
              id="DTM_NOW"
              key="DTM_NOW"
              name="application_id"
              onChange={[Function]}
              readOnly={true}
              type="radio"
              value="DTM_NOW"
            />
            <span
              className="test-assignment-ps__options"
            >
              Do the Math Now
            </span>
          </span>
        </div>
        <div
          className="test-assignment-ps__block-spacing"
        >
          <span>
             
            <SettingsSelectBox
              disabled={false}
              fieldClass="test-assignment-ps__select-field-disabled"
              fieldName="settingsSelectBox"
              fieldValue=""
              id=""
              label="2. Select a Module or Unit"
              labelClass="test-assignment-ps__select-label"
              labelPosition="left"
              mixedValue="2"
              onChange={[Function]}
            >
              <option
                disabled={true}
                hidden={false}
                value=""
              >
                 
              </option>
            </SettingsSelectBox>
          </span>
        </div>
        <div
          className="test-assignment-ps__select-test"
        >
          <span>
            <SettingsSelectBox
              disabled={false}
              fieldClass="test-assignment-ps__select-field-disabled"
              fieldName="settingsSelectBox"
              fieldValue=""
              id=""
              label="3. Select a Test:"
              labelClass="test-assignment-ps__select-label-test"
              labelPosition="left"
              mixedValue="2"
              onChange={[Function]}
            >
              <option
                disabled={true}
                hidden={false}
                value=""
              />
            </SettingsSelectBox>
          </span>
        </div>
      </div>
    </div>
    <div
      className="test-assignment-ps__footer"
    >
      <ProgramSettingsButtons
        saveAndReturnHandler={[Function]}
        setInitialValuesHandler={[Function]}
        showRestoreDefaults={false}
        stateResult={true}
        suppressSaveAndReturnRedirect={false}
      />
    </div>
  </div>
</form>
`;
